libs/base/Control/IOExcept.idr

```diff
@@ -10,8 +10,7 @@ instance Functor (IOExcept e) where
 
 instance Applicative (IOExcept e) where
      pure x = IOM (pure (pure x))
-     (IOM f) <*> (IOM a) = IOM (do f' <- f; a' <- a
-                                   return (f' <*> a'))
+     (IOM f) <*> (IOM a) = IOM $ (<*>) <$> f <*> a
```

Comment:
Yep. Wasn't sure which was preferred. I was worried that variant might not be so obvious that "it's defined by applying <*> to the inner in terms of the outer <*>" e.g. that pattern you see in the map and pure impls above. But it does look tidier.